src/ray/stats/metric_defs.cc

```diff
@@ -67,6 +67,15 @@ DEFINE_stats(actors,
              (),
              ray::stats::GAUGE);
 
+/// Track job by state, including RUNNING, FINISHED.
+DEFINE_stats(jobs,
+             "Current number of jobs currently in a particular state.",
+             // State: latest state for the particular job.
+             // JobId: ID in hex format for this job.
```

Comment:
Updated.

Reply:
Oh, sorry for the confusing. What I meant is that we still have one stats called `jobs` but have a `State` label with cardinality 2 (RUNNING and FINISHED). This will be stored as two time series in prometheus.

Reply:
Note: maybe you don't need to save all running_job_ids since you can keep both as a counter; and when the counter resets on gcs restart Prometheus can take care of the reset.

Reply:
> maybe you don't need to save all running_job_ids since you can keep both as a counter

Well you mentioned `HandleAddJob` needs to be idempotent (which I guess it could be invoked repeatedly due to reasons like retry), so we cannot increment/decrement the counter value directly.

The reason why I use set for running task count (gauge) and integer for finished task (counter) is:
- Gauge is queried on current value, it better to be precise (i.e. having 1 running job vs 0 running job differs a lot)
- Counter is queried via `rate` (as you mentioned above), which we care more about the rate of increment/decrement, rather than an absolute value